### PR TITLE
Add the score component

### PR DESCRIPTION
**Pull request by neshang**
**Added the score number + chart component**

A few changes:
1. I get rid of the tabs: As we are building a web app it might make more sense to display accent and clarity scores in two different component. So we can occupy more space horizontally...
2. I keep the current value input variable, which is the value of today T.

Screenshots on storybook:
Screenshot 2021-03-03 at 23 49 24

### CODE REVIEW

File: src/app/components/score-chart/score-chart.component.ts [hunk 1]
```diff
@@ -1,19 +1,64 @@
...
   selector: 'app-score-chart',
   templateUrl: './score-chart.component.html',
   styleUrls: ['./score-chart.component.sass']
 })
-export class ScoreChartComponent implements OnInit {
+export class ScoreChartComponent implements OnChanges {
 
-  private options: any = {};
+  @Input()
+  public historicalData: string[];
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.component.ts]: You should refine this type, as it is not clear what the data representation is like.
For example,

```typescript
interface Score {
    accentScore: number
    clarityScore: number
}

historicalData: Map<Date, Score>   // date as keys -> score as values
```

File: src/app/components/score-chart/score-chart.component.ts [hunk 2]
```diff
@@ -1,19 +1,64 @@
...
 
-  constructor() { }
+  @Input()
+  public label: string;
 
-  ngOnInit(): void {
-    Highcharts.chart('container', this.options);
-  }
+  @Input()
+  public currentValue = '- -';
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.component.ts]: I don't think we need default parameters (if TSLint is complaining, disable the flag?)
Moreover, the default value is of the wrong type (should be number?)

File: src/app/components/score-chart/score-chart.component.ts [hunk 3]
```diff
@@ -1,19 +1,64 @@
...
+        series: {
+            label: {
+                connectorAllowed: false
+            }
+        }
+    },
+    series: [],
+  };
+
+  ngOnChanges(changes: SimpleChanges): void {
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.component.ts:75]: I don't think we need this lifecycle hook:

1. `averageValue` can be a computed property (which will update itself when `historicalData` changes):
```typescript
get averageValue(): number {
  return mean(/* ... */)
}
```
2. `chartOptions` could be a property:
```typescript
get chartOptions() {
  return {
    /*...*/,
    series: [{
      /*...*/,
      data: this.historicalData./*...*/
    }]
  };
}
```

File: src/app/components/score-chart/score-chart.component.ts [hunk 4]
```diff
@@ -1,19 +1,64 @@
...
+    if (changes && changes.historicalData) {
+      this.averageValue = mean(changes.historicalData.currentValue.map(x => x[1]));
+      this.chartOptions.series = [{
+        name: 'Scores',
+        type: "line",
+        data: changes.historicalData.currentValue.map(x => [parseISO(x[0]), x[1]])
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.component.ts]: Better use pattern matching here to make it clear what the variables are:

```typescript
.map(([date, score]) => [parseISO(date), score])
```

File: src/app/components/score-chart/score-chart.stories.ts [hunk 5]
```diff
@@ -20,4 +24,51 @@ const Template: Story<ScoreChartComponent> = (args: ScoreChartComponent) => ({
...
+    },
+});
+
+
+export const WithCurrentValue = () => ({
+    component: ScoreChartComponent,
+    props: {
+        currentValue: 77,
+        label: 'Accent',
+        averageValue: mean(TestAccentScoreData.map(x => x[1])),
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.stories.ts]: For clarity:
```typescript
.map(([_, score]) => score)
```

File: src/app/components/score-chart/score-chart.component.ts [hunk 6]
```diff
@@ -1,19 +1,64 @@
...
+  public averageValue = '--';
+  public Highcharts: typeof Highcharts = Highcharts;
+  public chartOptions: Highcharts.Options = {
+    title: {
+      text: 'Track Your Progress'
+    },
+
+    yAxis: {
+        title: {
+            text: 'Score'
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.component.ts]: Not sure what score is this (accent? clarity?)

**Review by ansonmiu0214 — COMMENTED** [review]
General question: is this component rendering __both__ scores at the same time (accent on left, clarity over time on right)?

Or is this component rendering __one particular score__ (current accent score on left, accent score over time on right?)

I'm a bit confused because the line chart title doesn't show what score it is rendering

**Comment by ansonmiu0214** [comment]
@neshang please pull this branch locally before editing (fixed some conflicts)

**Comment by neshang** [comment]
@ansonmiu0214 not sure why there was conflict. I pulled from main and all the test passed in my last commit "Add test:prod command back".

I will take a look and make changes to resolve the comments today.

**Comment by ansonmiu0214** [comment]
Merge conflict was caused by merging PhraseDiff branch into main, and this branch was never merged with PhraseDiff branch. 

I've fixed the conflict, so you can pull your branch (score-chart) for the latest changes.

File: src/app/components/score-chart/score-chart.component.ts (hunk 1 above)
Comment (neshang) [reply src/app/components/score-chart/score-chart.component.ts]: Thats true. Will change!

File: src/app/components/score-chart/score-chart.component.ts (hunk 2 above)
Comment (neshang) [reply src/app/components/score-chart/score-chart.component.ts]: The TSLint is not really complaining, I just wanna show -- as the default value when there is no score for day T yet. But I guess I could also move the logic to the html file.

File: src/app/components/score-chart/score-chart.component.ts (hunk 3 above)
Comment (neshang) [reply src/app/components/score-chart/score-chart.component.ts:75]: That's true. I am just used to using lifecycle hook for variable changes in general. Feel like it is one single place for all the updates. but using get is more neat

File: src/app/components/score-chart/score-chart.component.ts (hunk 6 above)
Comment (neshang) [reply src/app/components/score-chart/score-chart.component.ts]: The label is used to show whether it is accent or clarity score. I can combine this with label as well to make it clearer

**Comment by neshang** [comment]
Re: (I couldn't reply directly to this comment somehow)

> General question: is this component rendering both scores at the same time (accent on left, clarity over time on right)?
> 
> Or is this component rendering one particular score (current accent score on left, accent score over time on right?)
> 
> I'm a bit confused because the line chart title doesn't show what score it is rendering

It is only going to render **one** score. We use label to know which score we are rendering

File: src/app/components/score-chart/score-chart.component.ts (hunk 6 above)
Comment (ansonmiu0214) [reply src/app/components/score-chart/score-chart.component.ts]: Yep, adding it in the graph (either title or axis or legend) would make it clearer.

File: src/app/components/score-chart/score-chart.component.ts (hunk 2 above)
Comment (ansonmiu0214) [reply src/app/components/score-chart/score-chart.component.ts]: I agree to move the "rendering logic" to the HTML file; `currentValue` can be an optional number, and the HTML file can render the "default value" (`--`) if no data is available today:

```typescript
@input
public currentValue?: number
```

**Comment by ansonmiu0214** [comment]
> Re: (I couldn't reply directly to this comment somehow)
> 
> > General question: is this component rendering both scores at the same time (accent on left, clarity over time on right)?
> > Or is this component rendering one particular score (current accent score on left, accent score over time on right?)
> > I'm a bit confused because the line chart title doesn't show what score it is rendering
> 
> It is only going to render **one** score. We use label to know which score we are rendering

Makes sense, understood!

File: tsconfig.json [hunk 7]
```diff
@@ -17,7 +17,8 @@
         "target": "es2015",
         "module": "es2020",
         "paths": {
-            "@components/*": ["app/components/*"]
+            "@components/*": ["app/components/*"],
```
Comment (neshang) [inline tsconfig.json:20]: Let's keep the paths like this.
It makes the imports look clean.

Comment (ansonmiu0214) [reply tsconfig.json:20]: Yep! (Didn't know why I/VS Code deleted it...)

File: src/app/components/score-chart/score-chart.component.ts (hunk 3 above)
Comment (neshang) [reply src/app/components/score-chart/score-chart.component.ts:75]: @ansonmiu0214 I tried using get with Highcharts options data, it seems like Highcharts will keep calling the get function and we end up calculating the data every time the get is called. This makes the rendering super slow and disable the tooltip function as well

**Review by ansonmiu0214 — APPROVED** [review]
LGTM :)

**Pull request merged**